lib/datadog/core/environment/container.rb

```diff
@@ -59,7 +59,7 @@ def descriptor
                                 || parts[-1][FARGATE_14_CONTAINER_REGEX, :container]
                 else
                   if (container_id = parts[-1][CONTAINER_REGEX, :container])
-                    task_uid = parts[-2][POD_REGEX, :pod]
+                    task_uid = parts[-2][POD_REGEX, :pod] || parts[1][POD_REGEX, :pod]
```

Comment:
Wow, is `parts[1][POD_REGEX, :pod]` a really fancy way of doing `POD_REGEX.match?(parts[1]) ? parts[1] : nil`? Always learning, definitely did now know you could do this.